src/build/config/parse.c

```diff
@@ -533,49 +507,41 @@ bldCfgParseOptionDeprecate(Yaml *const yaml)
         do
         {
             yamlEventCheck(optDeprecate, yamlEventTypeScalar);
-            BldCfgOptionDeprecateRaw optDeprecateRaw = {.name = optDeprecate.value, .reset = true};
+            const String *name = optDeprecate.value;
+            bool indexed = false;
 
             yamlEventNextCheck(yaml, yamlEventTypeMapBegin);
+            yamlEventNextCheck(yaml, yamlEventTypeMapEnd);
 
-            YamlEvent optDeprecateDef = yamlEventNext(yaml);
+            // Determine if this deprecation is indexed
+            const size_t questionPos = (size_t)strChr(name, '?');
 
-            if (optDeprecateDef.type == yamlEventTypeScalar)
+            if (questionPos != (size_t)-1)
             {
-                do
-                {
-                    yamlEventCheck(optDeprecateDef, yamlEventTypeScalar);
-                    YamlEvent optDeprecateDefVal = yamlEventNextCheck(yaml, yamlEventTypeScalar);
+                name = strNewFmt("%s%s", strZ(strSubN(name, 0, questionPos)), strZ(strSub(name, questionPos + 1)));
+                indexed = true;
+            }
 
-                    if (strEqZ(optDeprecateDef.value, "index"))
-                    {
-                        optDeprecateRaw.index = cvtZToUInt(strZ(optDeprecateDefVal.value));
-                    }
-                    else if (strEqZ(optDeprecateDef.value, "reset"))
-                    {
-                        optDeprecateRaw.reset = yamlBoolParse(optDeprecateDefVal);
-                    }
-                    else
-                        THROW_FMT(FormatError, "unknown deprecate definition '%s'", strZ(optDeprecateDef.value));
+            // Create final deprecation if it does not aready exist
+            BldCfgOptionDeprecateRaw *deprecate = lstFind(result, &name);
```

Comment:
Is the concept of having a `BldCfgOption*Raw` with a duplicate `BldCfgOption*` a holdover from Perl? It seems that some of these functions the `result` is the `BldCfgOption*` and some the `result` is `BldCfgOption*Raw` so it's a bit confusing when to use the Raw vs not-Raw.

Reply:
What implemented here is a two pass parser. Not everything is known in the first pass so the raw structs hold data in a simpler format until dependencies can be resolved. In some cases everything can be resolved up front and the raw struct is only used inside the function that parses the struct.